lib/src/task.dart

```diff
@@ -0,0 +1,59 @@
+class Task {
+  Task({
+    this.status,
+    this.updateId,
+    this.type,
+    this.duration,
+    this.enqueuedAt,
+    this.processedAt,
+    this.error,
+    this.details,
+  });
+
+  final String? status;
+  final int? updateId;
+  final String? type;
+  final String? duration;
+  final DateTime? enqueuedAt;
+  final DateTime? processedAt;
+  final UpdateError? error;
```

Comment:
I wonder if it is not better to rename this class `TaskError` or maybe just `Error`. What do you think?

Reply:
Good idea! :D